### ISSUE TITLE
Example proposal: validate image augmentation drift through metadata and embeddings

### ISSUE THREAD

**Issue by ternaus** [evidentlyai/evidently#1918]
# Example proposal

Would you be open to a focused cookbook example showing how to validate whether an image augmentation policy shifts a dataset beyond its intended bounds by using image metadata and, optionally, embeddings?

This would follow the boundary described in #887: Evidently would not consume raw images or add computer-vision-specific metrics. The example would convert the original and augmented image populations into the tabular properties or declared embeddings that Evidently already supports.

## User problem

An augmentation policy is intended to expand the training distribution, but it can become too aggressive: brightness may move outside the camera's plausible range, blur may remove useful detail, or combined transforms may shift image embeddings far from the source data. Looking at a few random previews does not quantify the population-level change, while discovering the problem from model performance is late and confounds data and training effects.

AlbumentationsX can generate a repeatable augmented view by applying the same resolved policy with stable per-sample `invocation_seed` values. Evidently can treat the original rows as the reference dataset and the augmented rows as the current dataset, then report and test distribution drift.

The user outcome would be a reproducible answer to: “Did this augmentation policy move the data farther than we intended?”

## Proposed workflow

The core notebook would avoid a dataset download and model-training loop. It would:

1. generate a deterministic population of varied RGB scenes;
2. define a bounded AX policy and an intentionally over-strong policy using transforms that affect luminance, contrast, and sharpness;
3. apply each policy with a stable `invocation_seed` derived from the sample identifier;
4. save `A.to_dict(policy)` so each report is tied to the configured augmentation policy;
5. extract one row of image properties per sample, such as mean luminance, RMS contrast, saturation, and Laplacian sharpness;
6. compare each augmented population with the original population using an Evidently `DataDriftPreset`; and
7. add pass/fail conditions that illustrate how a team can define its accepted augmentation envelope.

The report should make the contrast visible: a moderate policy stays within the chosen limits while the deliberately excessive policy fails one or more property-drift tests. The limits would be presented as tutorial values, not universal recommendations.

The Evidently boundary would remain a normal tabular workflow:

```python
from evidently import Report
from evidently.presets import DataDriftPreset

report = Report(
    [DataDriftPreset()],
    include_tests=True,
)

snapshot = report.run(
    augmented_properties,  # current
    original_properties,   # reference
)
```

## Optional embeddings extension

A short second section could show how users who already compute image embeddings can declare the embedding columns and run the current `EmbeddingsDrift` metric:

```python
from evidently import DataDefinition, Dataset, Report
from evidently.metrics import EmbeddingsDrift

definition = DataDefinition(
    embeddings={"image_embedding": embedding_columns},
)

original = Dataset.from_pandas(original_rows, definition)
augmented = Dataset.from_pandas(augmented_rows, definition)

embedding_report = Report(
    metrics=[EmbeddingsDrift(embeddings_name="image_embedding")],
)
embedding_snapshot = embedding_report.run(augmented, original)
```

The notebook would not download an embedding model by default. This section would accept precomputed columns or clearly remain optional, keeping the core example quick and self-contained.

## Why this fits Evidently

- **Evidently** owns the distribution metrics, Reports, and pass/fail tests over tabular metadata and declared embeddings.
- **AlbumentationsX** owns the controlled image transformation, stable sample-level realizations, and serialized policy provenance.
- **The user** can measure and constrain the data shift introduced by a proposed augmentation policy before training.

This implements the workaround recommended in #887 without claiming that Evidently supports raw images. It would require no new Evidently metric, preset, data type, or UI feature.

The example could live under `examples/cookbook` or in `evidentlyai/community-examples`, whichever location is easier for the maintainers to own.

## Dependency boundary

- The public `albumentationsx` package is AGPL-3.0-only and imports as `albumentations`.
- Importing it requires PyTorch. PyTorch is intentionally not selected by the package metadata because users need to choose the appropriate CPU, CUDA, or MPS build.
- AlbumentationsX would remain scoped to this example and would not become an Evidently runtime dependency.

I searched the current `evidentlyai/evidently` and `evidentlyai/community-examples` code, Issues, and pull requests and found no existing Albumentations or AlbumentationsX integration or overlapping augmentation-drift example.

If this use case fits the Evidently examples, I would be happy to implement and run the notebook after the maintainers confirm the preferred placement.

**Comment by LakshmirajSunilSawant** [comment]
Hi, I'd like to work on this if it's still open.
